[HBASE-22606] : BucketCache additional tests

virajjasani opened the pull request.

virajjasani commented:
@wchevreuil Please review. Due to build issue, had to close https://github.com/apache/hbase/pull/324 and create a new PR. Apologies for the same as I could not get why the build was giving -1 for patch even though the branch was already rebased with master.

saintstack commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java:
```diff
@@ -136,7 +135,7 @@ public void cacheBlock(BlockCacheKey cacheKey, Cacheable buf) {
   @Before
   public void setup() throws IOException {
     cache = new MockedBucketCache(ioEngineName, capacitySize, constructedBlockSize,
-        constructedBlockSizes, writeThreads, writerQLen, persistencePath);
```
Why make persistencePath null?

virajjasani replied:
Since persistencePath was defined as null and was not getting updated in the file ever after, I thought of removing the declaration and directly putting null

saintstack commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java:
```diff
@@ -258,17 +257,52 @@ public void testRetrieveFromFile() throws Exception {
 
+  @Test
+  public void testRetrieveFromMMap() throws Exception {
+    HBaseTestingUtility TEST_UTIL = new HBaseTestingUtility();
```
Do we get a new HBaseTestingUtility per test? If so, why not in the test setup  rather than at head of each test?
Also CAPITALIZATION of variables is for statics, not locals as here. 

Make a data member named testUtil and create the HTU instance in the @Before ?

virajjasani replied:
Done

saintstack commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java:
```diff
@@ -277,28 +311,49 @@ public void testRetrieveFromFile() throws Exception {
-    int[] smallBucketSizes = new int[] { 2 * 1024 + 1024, 4 * 1024 + 1024 };
+  @Test
+  public void testRetrieveFromFileWithoutPersistence() throws Exception {
+    HBaseTestingUtility TEST_UTIL = new HBaseTestingUtility();
```
Or is it that we sometimes create a new HTU? Not always? If so, we should be consistent... do one or the other...... but not a mix.

saintstack commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java:
```diff
@@ -277,28 +311,49 @@ public void testRetrieveFromFile() throws Exception {
     assertEquals(0, bucketCache.getAllocator().getUsedSize());
-    assertEquals(0, bucketCache.backingMap.size());
-
+    bucketCache.shutdown();
```
Should this be in a finally block?

virajjasani commented:
@saintstack Thanks for the review. Updated the PR. Please review

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 23 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 238 | master passed |
| +1 | compile | 54 | master passed |
| +1 | checkstyle | 64 | master passed |
| +1 | shadedjars | 258 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 207 | master passed |
| +1 | javadoc | 29 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 234 | the patch passed |
| +1 | compile | 52 | the patch passed |
| +1 | javac | 52 | the patch passed |
| +1 | checkstyle | 67 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 268 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 727 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 202 | the patch passed |
| +1 | javadoc | 32 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8393 | hbase-server in the patch passed. |
| +1 | asflicense | 23 | The patch does not generate ASF License warnings. |
| | | 11182 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/333 |
| JIRA Issue | HBASE-22606 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 7eaa99425f02 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 531d2902cb |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/2/testReport/ |
| Max. process+thread count | 4784 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

jatsakthi commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java:
```diff
@@ -255,51 +256,108 @@ public void run() {
-    Path testDir = TEST_UTIL.getDataTestDir();
-    TEST_UTIL.getTestFileSystem().mkdirs(testDir);
-
+    Path testDir = HBASE_TESTING_UTILITY.getDataTestDir();
```
These two lines for creation of test directory can be consolidated by usage of a private function to reduce code duplication, as there are multiple instances of these?

jatsakthi commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java:
```diff
@@ -322,13 +380,32 @@ public void testGetPartitionSize() throws IOException {
   }
 
+  @Test
+  public void testCacheSizeCapacity() throws IOException {
```
Let's leave a comment here saying "Test cache capacity (capacity / blockSize) < Integer.MAX_VALUE"?

jatsakthi commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java:
```diff
@@ -322,13 +380,32 @@ public void testGetPartitionSize() throws IOException {
+    }
+  }
+
+
```
nit: extra line.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 56 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 250 | master passed |
| +1 | compile | 56 | master passed |
| +1 | checkstyle | 68 | master passed |
| +1 | shadedjars | 278 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 204 | master passed |
| +1 | javadoc | 32 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 246 | the patch passed |
| +1 | compile | 56 | the patch passed |
| +1 | javac | 56 | the patch passed |
| +1 | checkstyle | 70 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 291 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 779 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 212 | the patch passed |
| +1 | javadoc | 35 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 17670 | hbase-server in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 20654 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.util.TestFromClientSide3WoUnsafe |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.regionserver.TestRegionMergeTransactionOnCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/333 |
| JIRA Issue | HBASE-22606 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 8a3ee482ac45 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 531d2902cb |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/1/testReport/ |
| Max. process+thread count | 4966 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

virajjasani commented:
> Apart from what stack mentioned, I have left few minor nits. Also what do you think of :
> 
> * Testing unsupported IOEngines as well (i.e. other then file/files, offheap, mmap, pmem)?
> * Testing more than one backing file? (using files:name1,name2 )?

@jatsakthi Thanks for your review. Updated the PR addressing all comments. Please take a look.
Review comments for changes suggested by @saintstack  are also updated.

Also, additional test for above mentioned scenarios are added:

- https://github.com/apache/hbase/pull/333/files#diff-ab0037c891480f9136f0e222ba490035R337
- https://github.com/apache/hbase/pull/333/files#diff-ab0037c891480f9136f0e222ba490035R350

jatsakthi commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java:
```diff
@@ -255,51 +269,128 @@ public void run() {
-    // persist cache to file
-    bucketCache.shutdown();
-    assertTrue(new File(persistencePath).exists());
+  @Test
```
@Test(expected = IllegalArgumentException.class) ?

jatsakthi replied:
Apart from this, the patch lgtm.

virajjasani replied:
It is removed so that we can specifically assert the IAE error message here: https://github.com/apache/hbase/pull/333/files#diff-ab0037c891480f9136f0e222ba490035R344

jatsakthi replied:
Perfect! +1

virajjasani replied:
Thanks a lot for your review @jatsakthi

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 46 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 235 | master passed |
| +1 | compile | 53 | master passed |
| +1 | checkstyle | 67 | master passed |
| +1 | shadedjars | 262 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 211 | master passed |
| +1 | javadoc | 31 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 228 | the patch passed |
| +1 | compile | 49 | the patch passed |
| +1 | javac | 49 | the patch passed |
| -1 | checkstyle | 63 | hbase-server: The patch generated 1 new + 15 unchanged - 0 fixed = 16 total (was 15) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 255 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 698 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 205 | the patch passed |
| +1 | javadoc | 31 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 19009 | hbase-server in the patch failed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 21780 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.procedure.TestSCPWithReplicas |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.replication.TestReplicationKillSlaveRSWithSeparateOldWALs |
|   | hadoop.hbase.quotas.TestSpaceQuotas |
|   | hadoop.hbase.replication.TestReplicationSmallTests |
|   | hadoop.hbase.replication.TestReplicationSmallTestsSync |
|   | hadoop.hbase.tool.TestBulkLoadHFiles |
|   | hadoop.hbase.master.TestSplitWALManager |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.tool.TestSecureBulkLoadHFiles |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.client.TestHbck |
|   | hadoop.hbase.util.TestFromClientSide3WoUnsafe |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/333 |
| JIRA Issue | HBASE-22606 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux dfd68492f8af 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / b42614295e |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/3/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/3/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/3/testReport/ |
| Max. process+thread count | 4825 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 166 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 286 | master passed |
| +1 | compile | 57 | master passed |
| +1 | checkstyle | 70 | master passed |
| +1 | shadedjars | 273 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 218 | master passed |
| +1 | javadoc | 32 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 237 | the patch passed |
| +1 | compile | 51 | the patch passed |
| +1 | javac | 51 | the patch passed |
| -1 | checkstyle | 65 | hbase-server: The patch generated 1 new + 15 unchanged - 0 fixed = 16 total (was 15) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 260 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 744 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 217 | the patch passed |
| +1 | javadoc | 34 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 16254 | hbase-server in the patch failed. |
| +1 | asflicense | 22 | The patch does not generate ASF License warnings. |
| | | 19309 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.client.TestMobSnapshotCloneIndependence |
|   | hadoop.hbase.security.access.TestSnapshotScannerHDFSAclController |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/333 |
| JIRA Issue | HBASE-22606 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 83df68984050 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 0198868531 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/4/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/4/testReport/ |
| Max. process+thread count | 4553 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 24 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 247 | master passed |
| +1 | compile | 52 | master passed |
| +1 | checkstyle | 67 | master passed |
| +1 | shadedjars | 268 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 203 | master passed |
| +1 | javadoc | 31 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 244 | the patch passed |
| +1 | compile | 53 | the patch passed |
| +1 | javac | 53 | the patch passed |
| -1 | checkstyle | 64 | hbase-server: The patch generated 1 new + 15 unchanged - 0 fixed = 16 total (was 15) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 260 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 738 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 224 | the patch passed |
| +1 | javadoc | 33 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 8096 | hbase-server in the patch failed. |
| +1 | asflicense | 24 | The patch does not generate ASF License warnings. |
| | | 10943 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.security.access.TestSnapshotScannerHDFSAclController |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/333 |
| JIRA Issue | HBASE-22606 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux b4367abd7f3a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 0198868531 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/5/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/5/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/5/testReport/ |
| Max. process+thread count | 4363 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/5/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

virajjasani commented:
@jatsakthi @busbey build is failing every time with some InterruptedIOException. Every time different test cases fail with IOE. How should we usually deal with it?

jatsakthi commented:
@virajjasani usually the best way is to check if the tests are present in the upstream flakey list. If yes, then we can usually bypass this. Else, you can run the failing tests locally to check if they ought to pass. Thirdly, if it's almost guaranteed that the test failure is unrelated to the changes proposed, in that case, we can ignore the tests failures.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 33 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 233 | master passed |
| +1 | compile | 50 | master passed |
| +1 | checkstyle | 65 | master passed |
| +1 | shadedjars | 251 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 204 | master passed |
| +1 | javadoc | 31 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 236 | the patch passed |
| +1 | compile | 53 | the patch passed |
| +1 | javac | 53 | the patch passed |
| +1 | checkstyle | 65 | the patch passed |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | shadedjars | 268 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 729 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 213 | the patch passed |
| +1 | javadoc | 34 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 7877 | hbase-server in the patch passed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 10673 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/333 |
| JIRA Issue | HBASE-22606 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 7e3d961e8ee9 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 0198868531 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/7/testReport/ |
| Max. process+thread count | 4298 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/7/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 49 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 277 | master passed |
| +1 | compile | 50 | master passed |
| +1 | checkstyle | 68 | master passed |
| +1 | shadedjars | 266 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 213 | master passed |
| +1 | javadoc | 33 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 258 | the patch passed |
| +1 | compile | 53 | the patch passed |
| +1 | javac | 53 | the patch passed |
| -1 | checkstyle | 67 | hbase-server: The patch generated 1 new + 15 unchanged - 0 fixed = 16 total (was 15) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 272 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 771 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 205 | the patch passed |
| +1 | javadoc | 31 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 16135 | hbase-server in the patch failed. |
| +1 | asflicense | 26 | The patch does not generate ASF License warnings. |
| | | 19100 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSide |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/333 |
| JIRA Issue | HBASE-22606 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux d357ba9bac9e 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 0198868531 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/6/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/6/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/6/testReport/ |
| Max. process+thread count | 4767 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-333/6/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

virajjasani commented:
> @virajjasani usually the best way is to check if the tests are present in the upstream flakey list. If yes, then we can usually bypass this. Else, you can run the failing tests locally to check if they ought to pass. Thirdly, if it's almost guaranteed that the test failure is unrelated to the changes proposed, in that case, we can ignore the tests failures.

Thanks @jatsakthi 
Yes test failures were irrelevant in this case. Anyways, the latest build is successful finally.
Could you please help me approve/merge this patch as per your latest review?
Thanks

virajjasani commented:
cc @Apache9 @busbey

virajjasani commented:
@wchevreuil @saintstack Please let me know if this looks good to you

virajjasani commented:
@Apache9 @busbey Could you please take a look

virajjasani commented:
Please review @saintstack @wchevreuil

wchevreuil commented:
lgtm +1 on the latest commits. @jatsakthi @saintstack let me know if you guys have any other remarks, if not, I can squash the commits and merge the PR into master with your "signedoffs".

virajjasani commented:
Please review @saintstack

jatsakthi commented:
The changes look good to me @wchevreuil

virajjasani commented:
Could you please take a look on this once @saintstack 
I would try to quickly update the PR in case of any concerns
